go/tasks/plugins/presto/utils/input_interpolator.go

```diff
@@ -0,0 +1,137 @@
+package presto
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/golang/protobuf/ptypes"
+	pb "github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/lyft/flytestdlib/logger"
+)
+
+// Matches any pair of open/close mustaches that contains a variable inside (e.g. `{{ abc }}`)
+var inputMustacheRegex = regexp.MustCompile(`{{\s*[^\s]+\s*}}`)
+
+// Matches the variable content inside a pair of double mustaches except for spaces and the mustaches themselves
+var inputVarNameRegex = regexp.MustCompile(`([^{{\s}}]+)`)
+
+// This will interpolate any input variable mustaches that were assigned to the statement, routingGroup, catalog, and
+// schema with the values of the input variables provided for the user.
+//
+// For example, if we have a Presto task like:
+//
+// presto_task = SdkPrestoTask(
+//    query="SELECT * FROM hive.city.fact_airport_sessions WHERE ds = '{{ds}}' LIMIT 10",
+//    output_schema=schema.schema_type,
+//    routing_group="{{ routing_group }}",
+//    catalog="hive",
+//    schema="city",
+//    task_inputs=inputs(ds=Types.String, routing_group=Types.String),
+// )
+//
+// Then  this function takes care of replace '{{ds}}' and '{{ routing_group }}' with the appropriate input values.
+func InterpolateInputs(
+	ctx context.Context,
+	inputs pb.LiteralMap,
+	routingGroup string,
+	catalog string,
+	schema string,
+	statement string) (string, string, string, string, error) {
+
+	inputsAsStrings, err := literalMapToStringMap(ctx, inputs)
+	if err != nil {
+		return "", "", "", "", err
+	}
+
+	// Remove implicit inputs from the rest of the inputs used for interpolation
+	delete(inputsAsStrings, "__implicit_routing_group")
+	delete(inputsAsStrings, "__implicit_catalog")
+	delete(inputsAsStrings, "__implicit_schema")
+
+	statement = interpolate(inputsAsStrings, statement)
+	routingGroup = interpolate(inputsAsStrings, routingGroup)
```

Comment:
actually, @EngHabu can you chime in here?  We only want to do statement right?  Everything else we just leave as is?

Reply:
Well, this approach would provide a bit more flexibility to users in terms of how much they are able to customize their Presto tasks. For example, if they want  their query to  only include  the table  name:
```
SELECT * FROM table_name
```
Then they could set the `catalog`  and  `schema` as inputs whose values perhaps come from a previous task and thus you wouldn't limit users to only be able to interpolate the query statement itself. Not sure  if interpolating only  the query will be enough for users. If so, then I can definitely simplify this. Your call.